tests-functional/tests/test_all_aboard_values.py

```diff
@@ -0,0 +1,41 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import pytest
+
+from selenium.webdriver.common.by import By
+from selenium.webdriver.support import expected_conditions as expected
+from selenium.webdriver.support.ui import WebDriverWait as Wait
+
+from pages.firstrun import Firstrun
+from pages.all_aboard import AllAboard
+
+
+@pytest.mark.nondestructive
+@pytest.mark.capabilities(firefox_args=['-foreground'])
+def test_first_sidebar_loads(base_url, selenium):
+    """ Tests that the sidebar opens after clicking the ActionButton"""
+    page = Firstrun(selenium, base_url).open()
+    page.complete_firstrun_values_flow()
+    # switch to browser chrome
+    selenium.set_context('chrome')
+
+    all_aboard = AllAboard(selenium)
+    # ensure that add-on action button is added as soon as we leave firstrun
+    Wait(selenium, 2).until(
+        expected.presence_of_element_located(
+            (By.ID, 'action-button--all-aboard-v1-all-aboard')
+        )
+    )
+
+    notification = Wait(selenium, 10).until(
+        expected.presence_of_element_located(
+            (By.CSS_SELECTOR, '#action-button--all-aboard-v1-all-aboard[badge="1"]')  # noqa E501
+        )
+    )
+
+    # once the first content notification happens, click the action button
+    notification.click()
+
+    assert all_aboard.sidebar.get_attribute('hidden') == 'false'
```

Comment:
Typically a lot of this would be taken care of in the page objects. I'm not familiar with the application here, but is 'all aboard' a page or a region within a page? The switching to chrome could also be inside the page/region object, and the test could look more like:

```python
page = Firstrun(selenium, base_url).open()
all_aboard = page.complete_firstrun_values_flow()
all_aboard.notification.click()
assert all_aboard.is_sidebar_displayed
```

Then you would return the `AllAboard` object from `complete_firstrun_values_flow` (I'm making an assumption here) and put the wait inside the `wait_for_page_to_load` for `AllAboard`. The `notification` could be a `Region` with a `wait_for_region_to_load`.